src/main/java/org/mondemand/Context.java

```diff
@@ -26,33 +28,25 @@ public String getKey() {
     return key;
   }
 
-  /**
-   * @param key the key to set
-   */
-  public void setKey(String key) {
-    this.key = key;
-  }
-
   /**
    * @return the value
    */
   public String getValue() {
     return value;
   }
 
-  /**
-   * @param value the value to set
-   */
-  public void setValue(String value) {
-    this.value = value;
-  }
-
-  public Context() {
-  }
-
-  public Context(String k, String v) {
+  public Context(String k, String v) throws MondemandException {
     key = k;
     value = v;
+    if(key == null) {
+      throw new MondemandException("key is null");
+    }
+    if(value == null || value.isEmpty()) {
```

Comment:
so based on what Anthony said in above, should I change this to 
```
if (value == null) {
  value = "";
}
```
?

Reply:
Depends, I guess we probably want to crash as that's more useful to the sender.  Otherwise they'd find out only after looking at the graphs which probably end up with a missing field.

Reply:
so we are good with the code as is in the PR.